mtx_skbuild_plugin.py

```diff
@@ -0,0 +1,90 @@
+"""
+This is a custom scikit-build-core plugin that will
+fetch the MaterialX version from the CMake project.
+"""
```

Comment:
@JeanChristopheMorinPerso Can these new root-scope files be moved into the `python` folder for clarity, or are there scikit limitations that would prevent this?  If there's a way to move them into the `python` folder, I think it would be a worthwhile refactor, as root-scope files tend to obscure the organization of a GitHub project as they accumulate.

Reply:
This file can be moved into a sub directory, but the `pyproject.toml` AFAIK. We could move it but it would break the source distribution (sdist) since installers and build tools expect the root of the sdist to contain a pyproject.toml (or a setup.py file). `scikit-build-core` doesn't provide a way to modify the sdist.

Reply:
Got it, and thanks for those additional details, @JeanChristopheMorinPerso.

Let's go ahead and move `mtx_skbuild_plugin.py` into the `python` folder for consistency, and it seems like a fair tradeoff for `pyproject.toml` to live in the root folder, given the major benefits that this new functionality will bring.

Reply:
Done!